app/main/views/briefs.py

```diff
@@ -194,6 +195,90 @@ def update_brief_status(brief_id):
     return jsonify(briefs=brief.serialize()), 200
 
 
+@main.route('/briefs/<int:brief_id>/<any(publish, withdraw):action>', methods=['PUT'])
+def update_brief_status_by_action(brief_id, action):
+    updater_json = validate_and_return_updater_request()
+
+    brief = Brief.query.filter(
+        Brief.id == brief_id
+    ).first_or_404()
+
+    if brief.framework.status != 'live':
+        abort(400, "Framework is not live")
+
+    action_to_status = {
+        'publish': 'live',
+        'withdraw': 'withdrawn'
+    }
+    if brief.status != action_to_status[action]:
+        brief.status = action_to_status[action]
+
+        validate_brief_data(brief, enforce_required=True)
+
+        audit = AuditEvent(
+            audit_type=AuditTypes.update_brief_status,
+            user=updater_json['updated_by'],
+            data={
+                'briefId': brief.id,
+                'briefStatus': brief.status,
+            },
+            db_object=brief,
+        )
+
+        db.session.add(brief)
+        db.session.add(audit)
+        db.session.commit()
+
+    return jsonify(briefs=brief.serialize()), 200
+
+
+@main.route('/briefs/<int:brief_id>/create-draft-from-withdrawn-brief', methods=['POST'])
+def create_draft_brief_from_withdrawn_brief(brief_id):
+    updater_json = validate_and_return_updater_request()
+
+    brief = Brief.query.filter(
+        Brief.id == brief_id
+    ).first_or_404()
+
+    if brief.framework.status != 'live':
+        abort(400, "Framework is not live")
+
+    if brief.status != 'withdrawn':
+        abort(400, "Brief status is not 'withdrawn'")
+
+    new_brief_data = brief.data
+    new_brief_data['title'] = 'Copy of ' + brief.data['title']
```

Comment:
I suggest we follow the approach we took with drafts and encapsulate this logic in the [model's `.copy` method](https://github.com/alphagov/digitalmarketplace-api/blob/master/app/models.py#L741). When it comes to the title change, drafts append `" copy"` instead of `"Copy of "` and only when the addition won't overflow the maxLength set for the title field by the schema (otherwise the validation step will fail). I think we should do the same here.